src/components/InputV2/Number.tsx

```diff
@@ -0,0 +1,84 @@
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+import { TextInput, TextInputProps } from './Text'
+
+export const NumberInput = ({
+  min,
+  max,
+  value,
+  onChange,
+  onBlur,
+  setHasError,
+  ...textProps
+}: NumberInputProps) => {
+  const [error, setError] = useState<string>('')
+  const onChangeText = (newTextValue: string) => {
+    setError('')
+    setHasError(false)
+    onChange(Number(newTextValue))
+  }
+  const onBlurText = (newTextValue: string) => {
+    const number = Number(newTextValue)
+    if (number < min) {
+      setError(`Has to be at least ${min}`)
+      setHasError(true)
+    }
+    if (number > max) {
+      setError(`Cannot be more than ${max}`)
+      setHasError(true)
+    }
+    onBlur(number)
+  }
+
+  return (
+    <TextInput
+      error={error}
+      type='number'
```

Comment:
Small nit - `TextInput` that takes a `type number` is interesting. We could follow the existing pattern and use some type of `BaseInput`?

Reply:
I see how that's semantically counterintuitive :)

The idea for TextInput was that it is the base for anything that looks like a html text input field, including the types `text`, `email`, `password`, `search` and `number`. But I see the merit of calling it `BaseInput` and implementing a separate `TextInput`, `PasswordInput`, etc